### PR TITLE
Add remote configuration client

### PR DESCRIPTION
**Pull request by lloeki**
**What does this PR do?**

Implement the remote configuration client class.

**Motivation**

Remote configuration

**Additional Notes**

Barebones implementation, some questions remain to be solved:

- there is AppSec-specific code in there (products and capabilities declaration)
- the callback system is missing because the repository does not yet report changes
- caching is not handled

There is an issue where the backend appears not to report any `client_configs` nor `target_files`, and `targets` may have targets but they may not correspond and/or include to the active client id.

In such gases, calling `client.sync` multiple times ends up getting the configuration. At this stage it is unknown whether this is expected API endpoint behaviour (e.g a delay in creating entries for a new client id, a "data lag" inside the agent...)

**How to test the change?**

```
  require 'ddtrace'

  Datadog.configure do |c|
    c.agent.host = '192.168.135.133'
    #c.tracing.enabled = true
    #c.diagnostics.debug = true
  end

  transport_options = {
    agent_settings: Datadog::Core::Configuration::AgentSettingsResolver.call(
      Datadog.configuration
    )
  }

  require 'datadog/core/transport/http'

  transport_v7 = Datadog::Core::Transport::HTTP.v7(**transport_options.dup)

  require 'datadog/core/remote/client'

  client = Datadog::Core::Remote::Client.new(transport_v7)
  repository = client.instance_eval { @repository }

  client.sync
  
  repository.contents
```

### CODE REVIEW

**Comment by lloeki** [comment]
Pending:

- specs
- RBS typing

**Comment by lloeki** [comment]
- Rebased using `git rebase --onto origin/add-remote-config-repository 414a6260c add-remote-config-client`
- Fixed usage of `ContentList#find` which was renamed to `#find_content`

**Comment by lloeki** [comment]
Rebased using `git rebase --onto add-remote-config-repository c9ef278c6^ add-remote-config-client`

**Comment by GustavoCaso** [comment]
Rebased using `git rebase master`

File: sig/datadog/core/remote/client.rbs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        def initialize: (Datadog::Core::Transport::Config::Transport transport, ?repository: Configuration::Repository) -> void
+
+        def sync: () -> void
+
+        class SyncError < StandardError
+        end
+
+        private
+
+        def payload: () ->  ::Hash[Symbol, untyped]
```
Comment (lloeki) [inline sig/datadog/core/remote/client.rbs:20]: I guess we could improve this `untyped` given that we expect only a specific set of types here.

**Review by GustavoCaso — APPROVED** [review]

File: sig/datadog/core/remote/client.rbs (hunk 1 above)
Comment (GustavoCaso) [reply sig/datadog/core/remote/client.rbs:20]: Sounds good. We can tackle that on a follow up PR

**Pull request merged**